### ISSUE TITLE
Button to download script as JSON

### ISSUE THREAD

**Issue by jthemphill** [tchajed/botc-tools#12]
I use [botc-tools](https://botc-tools.xyz/) to browse scripts, but once I've found a script, I want to export the script into [Pocket Grimoire](https://www.pocketgrimoire.co.uk/en_GB/). That requires me to download the script as JSON and reupload it into PG. Can we add a button to enable script downloading? I'm happy to write the code if you're on board and if you have a place where the button would look good. (Maybe next to the QR code?)

**Comment by tchajed** [comment]
I like this idea, but adding the button is actually the hard part. Adding one to the QR page seems like a good idea. I'd be happy to accept a PR adding one there.

I've also recently needed a way to get scripts into Pocket Grimoire, and indeed botc-tools is my favorite way to browse scripts. I think dealing with files is a bit of a pain on mobile - can you have the button copy a JSON array to the clipboard suitable for pasting? It's a format only useful for pocket grimoire but I can deal with that for now.

**Comment by tchajed** [comment]
I did a quick test and Pocket Grimoire does support the same format for pasting that the official script tool exports, something like this:

```json
[{"id":"_meta","name":"Reptiles II: Lizard in the City"},"chef","washerwoman","librarian","sailor","general","chambermaid","snake_charmer","flowergirl","undertaker","innkeeper","philosopher","fool","tea_lady","virgin","saint","sweetheart","klutz","drunk","barber","poisoner","devils_advocate","spy","psychopath","baron","al-hadikhia","storm_catcher","djinn"]
```

It's even permissive of "alhadikhia" (no hyphen) and "tealady" (no underscore) so you shouldn't have to get the character ids that the script tool uses (which are quite frankly a mess).

**Comment by tchajed** [comment]
This was fixed by #17.

**Comment by jw-taylor** [comment]
> This was fixed by [#17](https://github.com/tchajed/botc-tools/pull/17).

I found this feature here and wanted to use it too, I clicked the button but nothing seemed to happen (I was expecting it to download a json because I hadn't read the thread properly), tried a couple different browsers etc. before I actually read through everything here and realised it's a copy not a download.

It might be worth adding a toast message or something that says "Copied to clipboard!" when it's clicked, to clue in idiots like me :)

**Comment by tchajed** [comment]
I fully agree, #18 is a note to myself to add toast messages - it's not just you, even I get confused. I'm not much of a web developer so adding any sort of UI takes me a while...
